src/firecracker/tests/verify_dependencies.rs

```diff
@@ -0,0 +1,70 @@
+use cargo_toml::{Dependency, DependencyDetail, DepsSet, Manifest};
+use std::collections::HashMap;
+use std::path::Path;
+
+#[test]
+fn test_no_comparison_requirements() {
+    // HashMap mapping crate -> [(violating dependency, specified version)]
+    let mut violating_dependencies = HashMap::new();
+
+    let src_firecracker_path = std::env::var("CARGO_MANIFEST_DIR").unwrap();
+    let src_path = format!("{}/..", src_firecracker_path);
+
+    for fc_crate in std::fs::read_dir(src_path).unwrap() {
+        let fc_crate = fc_crate.unwrap();
+        if fc_crate.metadata().unwrap().is_dir() {
+            let violating_in_crate =
+                violating_dependencies_of_cargo_toml(fc_crate.path().join("Cargo.toml"));
+
+            if !violating_in_crate.is_empty() {
+                violating_dependencies.insert(
+                    fc_crate.file_name().into_string().unwrap(),
+                    violating_in_crate,
+                );
+            }
+        }
+    }
+
+    assert_eq!(
+        violating_dependencies,
+        HashMap::new(),
+        "No dependencies should be specified as comparison requirements. \
+        They should use caret requirements. See: \
+        https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html"
+    );
+}
+
+/// Parses the specified Cargo.toml file and returns any dependencies specified using a comparison requirements.
+///
+/// The return value maps the name of violating dependencies to the specified version
+fn violating_dependencies_of_cargo_toml(path: impl AsRef<Path>) -> HashMap<String, String> {
+    let manifest = Manifest::from_path(path).unwrap();
+    let mut violating = HashMap::new();
+
+    violating.extend(violating_dependencies_of_depsset(manifest.dependencies));
+    violating.extend(violating_dependencies_of_depsset(manifest.dev_dependencies));
+    violating.extend(violating_dependencies_of_depsset(
+        manifest.build_dependencies,
+    ));
+
+    violating
+}
+
+/// Returns an iterator over all dependencies in the given DepsSet specified using comparison requirements
+///
+/// The iterator produces tuples of the form (violating dependency, specified version)
+fn violating_dependencies_of_depsset(depsset: DepsSet) -> impl Iterator<Item = (String, String)> {
+    depsset
+        .into_iter()
+        .filter_map(|(name, dependency)| {
+            match dependency {
+                Dependency::Simple(version) => Some((name, version)),
+                Dependency::Detailed(DependencyDetail {
+                    version: Some(version),
+                    ..
+                }) => Some((name, version)),
+                _ => None, // doesn't tell us about version, e.g. if a direct git dependency is specified
+            }
+        })
+        .filter(|(_, version)| version.starts_with(">"))
+}
```

Comment:
```suggestion
fn violating_dependencies_of_depsset(depsset: DepsSet) -> impl Iterator<Item = (String, String)> {
    // Matches version numbers.
    let re = regex::Regex::new(r"[0-9]*\.[0-9]*\.[0-9]*");
    depsset.into_iter().filter_map(|(name, dependency)| match dependency {
            Dependency::Simple(version) | Dependency::Detailed(DependencyDetail { version: Some(version), ..}) 
                if !re.is_match(version) => Some((name, version)),
            _ => None, // doesn't tell us about version, e.g. if a direct git dependency is specified
    })
}
```

Reply:
I don't want to restrain non-comparison requirements, which is why I specifically match for those. Although I agree with your suggestions for simplifying the iterator :)